### ISSUE TITLE
breaking during printing causes Anomaly: main loop exited with exception: Sys.Break

### ISSUE THREAD

**Issue by JasonGross** [coq/coq#7140]
#### Version

8.8+beta1

#### Operating system

Linux

#### Description of the problem

Consider this command:
```bash
$ echo 'Require Import Coq.ZArith.ZArith Coq.Lists.List.
Goal True.
  pose (List.repeat Z.div_eucl 10) as k.
  Time vm_compute in k.' | coqtop -emacs -q & (export PID_KEEP_ALIVE=$!; sleep 1; echo done; kill -2 $PID_KEEP_ALIVE)
```
With Coq 8.7.1, it works fine.  With Coq 8.8+beta1, I see
```
Welcome to Coq 8.8+beta1 (March 2018)
<infomsg>Skipping rcfile loading.</infomsg>

<prompt>Coq < 1 || 0 < </prompt><infomsg>[Loading ML file z_syntax_plugin.cmxs ... done]</infomsg>
<infomsg>[Loading ML file quote_plugin.cmxs ... done]</infomsg>
<infomsg>[Loading ML file newring_plugin.cmxs ... done]</infomsg>
<infomsg>[Loading ML file omega_plugin.cmxs ... done]</infomsg>

<prompt>Coq < 2 || 0 < </prompt>1 subgoal (ID 1)

  ============================
  True

<prompt>Unnamed_thm < 3 |branch| 0 < </prompt>1 subgoal (ID 4)

  k := repeat Z.div_eucl 10 : list (Z -> Z -> Z * Z)
  ============================
  True

<prompt>Unnamed_thm < 4 |branch| 0 < </prompt><infomsg>Finished transaction in 0.156 secs (0.135u,0.02s) (successful)
</infomsg>
done
Anomaly: main loop exited with exception: Sys.Break
Please report at http://coq.inria.fr/bugs/.
<prompt>Unnamed_thm < 5 |branch| 0 < </prompt>
```
(note that the `done` is from my `echo` right before the SIGINT)

cc @gares @ejgallego

**Comment by ejgallego** [comment]
This is already fixed in `master`, so we need an 8.8-specific fix.

**Comment by maximedenes** [comment]
@Zimmi48 has a fix been backported?

**Comment by ejgallego** [comment]
The fix is in #7143 I think.

**Comment by Zimmi48** [comment]
Yep, sorry, I forgot to close. GitHub auto-close doesn't work when we don't merge in `master` (because they don't distinguish feature branches and stable branches).
